bugbug/similarity.py

```diff
@@ -93,24 +93,58 @@ def evaluation(self):
         total_p = 0
         hits_p = 0
 
+        recall_rate_1 = 0
+        recall_rate_5 = 0
+        recall_rate_10 = 0
+        precision_rate_1 = 0
+        precision_rate_5 = 0
+        precision_rate_10 = 0
+        preckq = 0
+        queries = 0
+
         for bug in bugzilla.get_bugs():
             if duplicates[bug["id"]]:
+
+                queries += 1
                 similar_bugs = self.get_similar_bugs(bug)
 
                 # Recall
                 for item in duplicates[bug["id"]]:
                     total_r += 1
                     if item in similar_bugs:
                         hits_r += 1
+                    if item in similar_bugs[:1]:
+                        recall_rate_1 += 1
+                    if item in similar_bugs[:5]:
+                        recall_rate_5 += 1
+                    if item in similar_bugs[:10]:
+                        recall_rate_10 += 1
 
                 # Precision
-                for element in similar_bugs:
+                for idx, element in enumerate(similar_bugs):
                     total_p += 1
                     if element in duplicates[bug["id"]]:
                         hits_p += 1
+                        if idx == 1:
+                            precision_rate_1 += 1
+
+                        if idx in list(range(5)):
+                            precision_rate_5 += 1 / 5
+
+                        if idx in list(range(10)):
+                            precision_rate_10 += 1 / 10
+
+                        preckq += 1 / ((idx + 1) * len(similar_bugs))
 
+        print(f"Recall @ 1: {recall_rate_1/total_r * 100}%")
```

Comment:
I think the code would be more easy to understand if we treated recall_rate like precision_rate (so calculating a recall for each query and then dividing by the number of queries)